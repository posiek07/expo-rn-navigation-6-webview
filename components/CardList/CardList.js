import React from 'react';
import styled from 'styled-components/native';
import Card from '../Card/Card';
import propTypes from 'prop-types';

const StyledSafeAreaView = styled.SafeAreaView`
  background-color: white;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  flex: 1;
`;

const StyledFlatList = styled.FlatList`
  display: flex;
`;

const CardList = ({events}) => (
  <StyledSafeAreaView>
    {!!events && (
      <StyledFlatList
        data={events}
        renderItem={props => <Card {...props} />}
        keyExtractor={item => item.url} // need to be unique (id's are the same in database)
      />
    )}
  </StyledSafeAreaView>
);

CardList.propTypes = {
  events: propTypes.array,
};

export default CardList;
