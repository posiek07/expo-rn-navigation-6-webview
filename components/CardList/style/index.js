import styled from 'styled-components/native';

export const StyledSafeAreaView = styled.SafeAreaView`
  background-color: white;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  flex: 1;
`;

export const StyledFlatList = styled.FlatList`
  display: flex;
`;
