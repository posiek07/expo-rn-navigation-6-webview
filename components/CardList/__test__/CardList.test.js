import CardList from '../CardList';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';
import data from '../../../test-data/data.json';

it('CardList should render and match snapshot', async () => {
  const events = data.events;
  const cardList = renderer.create(<CardList item={events} />).toJSON();
  expect(cardList).toMatchSnapshot();
});
