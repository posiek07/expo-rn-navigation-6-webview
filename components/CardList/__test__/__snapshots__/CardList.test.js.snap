// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList should render and match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
/>
`;
